From the failing CI:
```
D:\a\Nim\Nim\lib\pure\terminal.nim(226, 24) Error: getCursorPos(getCurrentProcess()) can raise an unlisted exception: ref OSError
FAILURE
Error: Process completed with exit code 1.
```

Super sophisticated test program:
```nim
# test.nim
import terminal
echo getCursorPos()
```

Placing it in the same dir as terminal.nim and running `nim c test.nim` replicates the error, and it goes away after adding `OSError` to the list.  The test program fails, however:
```
PS C:\Users\matt\src\Nim\lib\pure> .\test.exe
C:\Users\matt\src\Nim\lib\pure\test.nim(2) test
C:\Users\matt\src\Nim\lib\pure\terminal.nim(225) getCursorPos
C:\Users\matt\src\Nim\lib\pure\terminal.nim(221) getCursorPos
C:\Users\matt\.choosenim\toolchains\nim-2.0.0\lib\std\oserrors.nim(92) raiseOSError
Error: unhandled exception: The handle is invalid.
 [OSError]
```

Fixed it like so: 
```nim
  proc getCursorPos*(): tuple [x, y: int] {.raises: [ValueError, IOError, OSError].} =
    return getCursorPos(getStdHandle(STD_OUTPUT_HANDLE))
```

Tested on windows, works ok.

--- a/lib/pure/terminal.nim
+++ b/lib/pure/terminal.nim
@@ -221,9 +221,8 @@ when defined(windows):
       raiseOSError(osLastError())
     return (int(c.dwCursorPosition.x), int(c.dwCursorPosition.y))
 
-    # TODO: test this
-  proc getCursorPos*(): tuple [x, y: int] {.raises: [ValueError, IOError].} =
-    return getCursorPos(getCurrentProcess())
+  proc getCursorPos*(): tuple [x, y: int] {.raises: [ValueError, IOError, OSError].} =
+    return getCursorPos(getStdHandle(STD_OUTPUT_HANDLE))
 
   proc setCursorPos(h: Handle, x, y: int) =
     var c: COORD

--- a/lib/pure/terminal.nim
+++ b/lib/pure/terminal.nim
@@ -273,6 +273,7 @@ else:
 
   proc getCursorPos*(): tuple [x, y: int] {.raises: [ValueError, IOError].} =
     ## Returns cursor position (x, y)
+    ## writes to stdout and expects the terminal to respond via stdin
     var
       xStr = ""
       yStr = ""

